gen2-record-replay/libraries/depthai_record.py

```diff
@@ -242,3 +270,193 @@ def stream_out(name, fps, out, noEnc=False):
+        # http://docs.ros.org/en/noetic/api/sensor_msgs/html/index-msg.html
+        schema_id = self.writer.register_schema(
+            "ros.sensor_msgs.PointCloud2",
+            "jsonschema",
+            data=json.dumps(
+                {
+                    "type": "object",
+                    "properties": {
+                        "header": {
+                            "type": "object",
+                            "properties": {
+                                "seq": {"type": "integer"},
+                                "stamp": {
+                                    "type": "object",
+                                    "properties": {
+                                        "sec": {"type": "integer"},
+                                        "nsec": {"type": "integer"},
+                                    },
+                                },
+                                "frame_id": {"type": "string"}
+                            },
+                        },
+                        "height": {"type": "integer"},
+                        "width": {"type": "integer"},
+                        "fields": {
+                            "type": "array",
+                            "items": {
+                                "type": "object",
+                                "properties": {
+                                    "name": {"type": "string"},
+                                    "offset": {"type": "integer"},
+                                    "datatype": {"type": "integer"},
+                                    "count": {"type": "integer"}
+                                }
+                            },
+                        },
+                        "is_bigendian": {"type": "boolean"},
+                        "point_step": {"type": "integer"},
+                        "row_step": {"type": "integer"},
+                        "data": {"type": "string", "contentEncoding": "base64"},
+                        "is_dense": {"type": "boolean"}
+                    },
+                },
+            ).encode("utf8")
+        )
+
+        # create and register channel
+        self.channel_id = self.writer.register_channel(
+            schema_id=schema_id,
+            topic="pointClouds",
+            message_encoding="json",
+        )
+
+    # send in image read with "cv2.getCvFrame"
+    def imageSave(self, img):
+        # convert cv2 image to .jpg format
+        # is_success, im_buf_arr = cv2.imencode(".jpg", img)
+
+        # read from .jpeg format to buffer of bytes
+        # byte_im = im_buf_arr.tobytes()
+        byte_im = img.tobytes()
+
+        # data must be encoded in base64
+        data = base64.b64encode(byte_im).decode("ascii")
+
+        tmpTime = time.time_ns()
+        sec = math.trunc(tmpTime / 1e9)
+        nsec = tmpTime - sec
+
+        self.writer.add_message(
+            channel_id=self.channel_id,
+            log_time=tmpTime,
```

Comment:
```suggestion
            log_time=time.time_ns(),
```

Reply:
This is needed, as the time must be same as time in header (calling it 2 times produces 2 different times.